### DISCUSSION TITLE
How to have Fixed IP?

### DISCUSSION THREAD

**Discussion by albarreto** [devmfc/debian-on-amlogic#82] [Q&A]
Hi folks,

My newest router is locked by provider and there's no option for lock ip by MAC.
I have TX3 Mini+ with my docker server and I need change IP for a fixed.
At first, I tried install Cockpit for change IP using a GUI. But when I reboot, the ip number come back to provided by DHCP

I tried to use nmcli commands:
```
# nmcli c mod 'eth0' ipv4.addresses 192.168.0.24/24 ipv4.method manual
# nmcli con mod 'eth0' ipv4.gateway 192.168.0.1
# nmcli con mod 'eth0' ipv4.dns "192.168.0.1"
```

The IP is changed, but when I reboot, the older IP backs.
After reboot, it is a "nmcli c show" command:

```
NAME                UUID                                  TYPE      DEVICE
eth0                6b9bd335-2505-4fcc-b85e-4b32ce1015bf  ethernet  eth0 <=== Recreated after boot
br-73de9cc8efd0     354a44d6-4f1e-4e33-8529-1c267f262c12  bridge    br-73de9cc8efd0
docker0             4a6edebd-15f0-4f7a-ba61-d90d1b9cc870  bridge    docker0
lo                  a618d8ef-3fca-4cc7-b43f-6ef16608f7b8  loopback  lo
br-3c1a15e4a19c     a9f5f40c-97ce-4e36-a6f9-37ac8bb0df7b  bridge    br-3c1a15e4a19c
br-4bf9f8a99221     6409829c-5acc-46b9-b6d2-1936e43064c6  bridge    br-4bf9f8a99221
eth0                9de642e9-a289-4974-87aa-1bd0fe9f0101  ethernet  --
```

Any idea?

**Comment by ghost** [comment]
uhm type this command and see output....

sudo nmcli connection



...name "eth0" is strange for a wired lan...in all my debian based is "Wired connection 1" like the pic...


![nmcli-connection-command-debian12](https://github.com/user-attachments/assets/c6b82ba1-39ce-4e5a-b94a-2abbb73730c1)

so, basically, the command is: 

sudo nmcli connection modify 'Wired connection 1' ipv4.address your-static-ip/24
sudo nmcli connection modify 'Wired connection 1' ipv4.gateway ip-of-your-gateway/router
sudo nmcli connection modify 'Wired connection 1' ipv4.method manual
sudo nmcli connection modify 'Wired connection 1' ipv4.dns '8.8.8.8'
sudo reboot

replace "your-static-ip" with your ip and "ip-of-your-gateway/router" with router/gateway ip....
(8.8.8.8 is only an example...here you need to put your primary DNS IP)

**Comment by albarreto** [comment]
Hi, thanks for your reply.

I tried your suggestion, but is didn't work.
I typed (without errors):
```
sudo nmcli connection modify 'Wired connection 1' ipv4.address 192.168.0.24/24
sudo nmcli connection modify 'Wired connection 1' ipv4.gateway 192.168.0.1
sudo nmcli connection modify 'Wired connection 1' ipv4.method manual
sudo nmcli connection modify 'Wired connection 1' ipv4.dns '8.8.8.8'
sudo reboot
```

The IP remains 192.168.0.137 when it starts.

```
# sudo nmcli connection
NAME                UUID                                  TYPE      DEVICE
eth0                f72c913b-b181-4ac1-bedf-1fde60a8134c  ethernet  eth0
br-73de9cc8efd0     259d9857-96e9-45fa-9c21-d5ed57840ed6  bridge    br-73de9cc8efd0
docker0             4f7c53d4-8188-4d47-b678-e9eae50fc22f  bridge    docker0
lo                  ee521d9b-3a0b-4acb-b7c3-312d114a2723  loopback  lo
br-3c1a15e4a19c     f99e8e72-4fee-44c5-8aa0-e2a4897d76fc  bridge    br-3c1a15e4a19c
br-4bf9f8a99221     1b41f819-088c-4303-989d-3ca7f5f106b5  bridge    br-4bf9f8a99221
eth0                9de642e9-a289-4974-87aa-1bd0fe9f0101  ethernet  --
eth0                6f5201da-263f-471f-a15b-35ab97a32e40  ethernet  --
Wired connection 1  5b8d34ec-4e2c-3173-8b76-b67fc398671e  ethernet  --
```

BTY, when I tried the original commands (using eth0), I lost ssh connection and a network scanner app shows the updated IP, but I reboot, the ip 137 returns.

NOTE: I'm using a tv box Tanix TX3 Mini+
```
uname -a
Linux docker-tx3miniplus 6.6.8-meson64 #16 SMP PREEMPT Fri Dec 22 10:24:05 CET 2023 aarch64 GNU/Linux

```

**Reply by ghost** [reply]
you have 4 eth port configured in your box that have only one eth LoL ...i think the more convenient way is to reinstall debian (or ubuntu) ...my knowledge stop here ;\ i'm very sorry

**Comment by albarreto** [comment]
No problem.  👍 
That additionals ports were added afer each nmcli command.
OS adds and changes UUID

**Comment by gleitoncampos** [comment]
In my case, this debian 12 image uses Systemd-Networkd as network manager.  Check this documentation, it worked to me: https://wiki.debian.org/SystemdNetworkd
